File: find_english.php
<?php
$file = $argv[1];
$lines = file($file);
$count = 0;

foreach ($lines as $i => $line) {
    $line = trim($line);
    if (empty($line) || strpos($line, '//') === 0) continue;
    
    if (preg_match("/^\$_LANG\['([^']+)'\]\s*=\s*'([^']*)';$/", $line, $m)) {
        $key = $m[1];
        $val = $m[2];
        
        // Skip if contains Arabic characters
        if (preg_match('/[\x{0600}-\x{06FF}]/u', $val)) continue;
        
        // Skip empty values
        if (empty($val)) continue;
        
        // Skip pure numbers/symbols
        if (preg_match('/^[\d\s\-\.:\/,\(\)]+$/', $val)) continue;
        
        // Skip very short strings
        if (strlen($val) < 3) continue;
        
        // Skip standard English-only patterns
        if (preg_match('/^[A-Z][a-z]+(\s[A-Z][a-z]+)*\s*$/', $val) 
            || preg_match('/^[a-z]/', $val)) {
            echo "Line " . ($i+1) . ": " . $line . "\n";
            $count++;
        }
    }
}
echo "\nTotal English strings: $count\n";

File: find_english.py
#!/usr/bin/env python3
import re, sys

filepath = sys.argv[1]
lines = open(filepath).readlines()
count = 0

for i, line in enumerate(lines):
    line = line.strip()
    if not line or line.startswith('//'):
        continue
    m = re.match(r"^\$_LANG\['([^']+)'\]\s*=\s*'([^']*)';$", line)
    if m:
        key, val = m.groups()
        if re.search(r'[\u0600-\u06FF]', val):
            continue
        if not val or len(val) < 3:
            continue
        if re.match(r'^[\d\s\-\.:\/,\(\)\@\+]+$', val):
            continue
        if re.match(r"^[A-Za-z][\sA-Za-z0-9,.\'\"()!?&%@#\/\-:;]+$", val):
            print(f"Line {i+1}: {line}")
            count += 1

print(f"\nTotal English strings: {count}")
